src/components/SelectionPane.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import styled from 'styled-components'
+
+import CollapsableHeader from './CollapsableHeader'
+import SelectionCard from './SelectionCard'
+
+import {getFaction} from 'util/jsonUtils'
+
+
+const Body = styled.div`
+    border-style: solid;
+    border-width: 1px;
+    flex-grow: 1;
+    border-right: 0
+`;
+
+
```

Comment:
I hate the render on this, it works but is hard to read, looking for suggestions on how to break it into method(s)

Reply:
It's actually not too bad. One thing that'll clean them up a lot are implicit returns in your `map` calls. At that point it's probably clean enough to leave it here I think.

```
return (
        <Body>
            {Object.keys(curData).map((item, index) => (
            <CollapsableHeader key={item} title={item}>
                {
                    curData[item].map((subitem, i) => (
                        <SelectionCard
                            unit={subitem}
                            key={subitem.name}
                        />
                    ))
                }
            </CollapsableHeader>
            )
        )}
        </Body>
    )
```

Otherwise you could move the inner selection card code to another function:
```
const renderSelectionCards = (data) => (
    <>
        {data.map((subitem, i) => (
            <SelectionCard
                unit={subitem}
                key={subitem.name}
            />
        )}
    </> //shorthand for React.Fragment btw
);

//in your render return
//...

return (
        <Body>
            {Object.keys(curData).map((item, index) => (
            <CollapsableHeader key={item} title={item}>
                {renderSelectionCards(curData[item])}
            </CollapsableHeader>
            )
        )}
        </Body>
    )
```

Reply:
👌

Reply:
It'd also probably be a readability boost break down `curData` a bit! Following Chris's notes, could look like:

```
const UnitSection = ({ title, units }) => (
    <CollapsableHeader>
        ...
    </CollapsableHeader>
);

const SelectionPane = (props) => {
    const curData = getFaction('rebel');
    const sections = Object.keys(curData).map(key => {
        title: key,
        units: curData[key] //these appear to be "commanders", "corps", etc. ya?
    });
    return (
        <Body>
            {sections.map(section => <UnitSection header={section.title} ... />)
        </Body>
    )
};
```

Reply:
Took alot of the suggestions here! looks alot cleaner :D. Will probably end up moving it out into a separate comp later tho :D